addons/binding/org.openhab.binding.boschindego/src/main/java/org/openhab/binding/boschindego/handler/BoschIndegoHandler.java

```diff
@@ -0,0 +1,249 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.boschindego.handler;
+
+import static org.openhab.binding.boschindego.BoschIndegoBindingConstants.*;
+
+import java.math.BigDecimal;
+
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import de.zazaz.iot.bosch.indego.DeviceCommand;
+import de.zazaz.iot.bosch.indego.DeviceStateInformation;
+import de.zazaz.iot.bosch.indego.DeviceStatus;
+import de.zazaz.iot.bosch.indego.IndegoAuthenticationException;
+import de.zazaz.iot.bosch.indego.IndegoController;
+import de.zazaz.iot.bosch.indego.IndegoException;
+
+/**
+ * The {@link BoschIndegoHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Jonas Fleck - Initial contribution
+ */
+public class BoschIndegoHandler extends BaseThingHandler {
+
+    private Logger logger = LoggerFactory.getLogger(BoschIndegoHandler.class);
+    private int commandToSend;
```

Comment:
IMHO this should rather be a thread-safe queue, so that you will only need to send a command if you find something in there. That would eliminate the need for synchronization and is by far less error prone